### PR TITLE
Automate OpenCollective PR notifications

### PR DESCRIPTION
**Pull request by mikehardy**
## Purpose / Description

I'm the only person doing OpenCollective monthly notices and payment processing and frankly I was so bogged down I couldn't get it done. It takes maybe a half hour a time to do it, and it backed up about 4 months which means I had a couple hours of budget to automate it, so here we are

## Fixes
Fixes no issue, just a manual process

## Approach
Inspired by @krmanik's fabulous work in actions, I know I can use javascript in actions now with the github API. Pretty easy really

## How Has This Been Tested?

26 different commits + pushes + workflow runs, building it piece by working piece because the github workflow developer experience is frankly awful, but whatever

final test https://github.com/mikehardy/Anki-Android/actions/runs/3079799321/jobs/4976437491
test result https://github.com/mikehardy/Anki-Android/pull/105#issuecomment-1250538408

## Learning (optional, can help others)
query for pulls https://docs.github.com/en/rest/search#search-issues-and-pull-requests for the query
script action https://github.com/actions/github-script
how to get inputs in a workflow using javascript https://stackoverflow.com/a/70999557/9910298

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by mikehardy** [comment]
The way to run this thing is with inputs like
startDate 2022-05-01
endDate 2022-05-31

I'm going to do it and see how it goes...

**Comment by mikehardy** [comment]
would love comments though from anyone else that sees this - will do followups

**Comment by mikehardy** [comment]
A thing of beauty :-)
https://github.com/ankidroid/Anki-Android/actions/runs/3079838583/jobs/4976519821
image

File: .github/workflows/opencollective_notices.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+                      Please understand that our monthly budget is never guaranteed to cover all claims - the cap on payments-per-person may be lower, but we try to make our process as fair and transparent as possible, we just need your understanding.
+
+                      Thanks!`
+                        })
+                      }
+
+
+                      // Now get the PRs for the month, iterating over pages if needed (100 per page limit)
+                      let result = await getPRPage(startDate, endDate, 1)
```
Comment (david-allison) [inline .github/workflows/opencollective_notices.yml:61]: Could you note that pages are 1-indexed (if they are?)

File: .github/workflows/opencollective_notices.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                        // skip some users
+                        if (key === 'github-actions' || key.includes('dependabot')) {
+                          console.log(`ignoring pr with bot user ${key}`);
+                          continue;
+                        }
+                        console.log(`Going to notify ${key} on PR ${uniqueContributorsWithPR[key]}`);
+                        try {
+                          await addComment(uniqueContributorsWithPR[key], key);
+                        } catch (e) {
+                          console.log('what happend?' + e);
```
Comment (david-allison) [inline .github/workflows/opencollective_notices.yml:92]: can we collect the errors and fail if one exists?

**Review by david-allison — COMMENTED** [review]
Looks great

File: .github/workflows/opencollective_notices.yml (hunk 2 above)
Comment (mikehardy) [reply .github/workflows/opencollective_notices.yml:92]: Of course! quick and dirty is nice for a first run (successful!) but robustness wins the race. Good suggestion

File: .github/workflows/opencollective_notices.yml (hunk 1 above)
Comment (mikehardy) [reply .github/workflows/opencollective_notices.yml:61]: indeed they are, I can link to the pagination docs in fact

File: .github/workflows/opencollective_notices.yml (hunk 2 above)
Comment (mikehardy) [reply .github/workflows/opencollective_notices.yml:92]: param validation would be nice as well - on the robustness side, perhaps I can include a note / example in the workflow dropdown 🤔 
In the future I'll add the ability run from schedule, schedule it for 1st of the month, and have it determine the date range automagically anyway

**Comment by mikehardy** [comment]
Just to string things together for anyone following along at home - the follow-on with review comments is up #12447

**Pull request merged**